llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -127,6 +130,8 @@ class MachineSinking : public MachineFunctionPass {
   const MachineBranchProbabilityInfo *MBPI = nullptr;
   AliasAnalysis *AA = nullptr;
   RegisterClassInfo RegClassInfo;
+  Pass *LegacyPass;
+  MachineFunctionAnalysisManager *MFAM;
```

Comment:
Keep pass manager details out of the implementation

Reply:
The other alternative is passing all analyses as parameters and five seemed too many

Reply:
That is what you should do, and what other passes do

Reply:
@optimisan this is pending.